Add ListExtension Method SelectWhereNot

TheDioBrando opened the pull request:
Add Func funcCatColorIsDark
Add Func funcPersonPhoneNumberStartsWith7

TheDioBrando committed:
Add ListExtension Method SelectWhereNot
Add Func funcCatColorIsDark
Add Func funcPersonPhoneNumberStartsWith7

TIove commented on src/FuncHW/ListExtensions.cs:
```diff
@@ -12,5 +12,18 @@ public static class ListExtensions
 
         //add extension for List SelectWhereNot: receives Func, returns the list of elements that don't match the condition
+        public static List<T> SelectWhereNot<T>(this List<T> values, Func<T, bool> func)
+            where T : class
```
For what this condition? What if I want to use this extension with 'int' or other struct?

TheDioBrando replied:
Delete this condition

TIove commented on src/FuncHW/ListExtensions.cs:
```diff
@@ -12,5 +12,18 @@ public static class ListExtensions
+
+            foreach (var item in values)
+            {
+                if(!func(item))
```
Space after if. Should be `if ()`.

TIove replied:
Also, that's not so bad imho, but in DigitalOffice we write always with brackets `{ }`

TheDioBrando replied:
Done

TIove commented on src/FuncHW/Program.cs:
```diff
@@ -27,11 +27,18 @@ static void Main()
             //Func*** funcPersonIsActive = ***;
             //Func*** funcPersonPhoneNumberStartsWith7 = ***;
+            Func<Person, bool> funcPersonPhoneNumberStartsWith7 = (Person person) => 
+                person.PhoneNumber / 10000 == 7;
```
No, that's bad to use `/ 10000`. What if appear new phone number with six symbols? e.x. 123456. 
Try to come up with something else.

TheDioBrando replied:
Remake check for any length of number.

TIove commented on src/FuncHW/Program.cs:
```diff
@@ -27,11 +27,18 @@ static void Main()
 
             //Func*** funcCatIsDomestic = ***;
             //Func*** funcCatColorIsDark = ***;
+            Func<Cat, bool> funcCatColorIsDark=(Cat cat)=> 
```
Be careful with spaces. `...Dark = (Cat cat) =>`

TheDioBrando replied:
Done

TIove commented on src/FuncHW/Program.cs:
```diff
@@ -27,11 +27,18 @@ static void Main()
             //Func*** funcCatIsDomestic = ***;
             //Func*** funcCatColorIsDark = ***;
+            Func<Cat, bool> funcCatColorIsDark=(Cat cat)=> 
+                string.Equals(cat.Color, "grey",StringComparison.OrdinalIgnoreCase) 
```
Add space after comma.

TheDioBrando replied:
Done

TIove commented on src/FuncHW/Program.cs:
```diff
@@ -27,11 +27,18 @@ static void Main()
+
             //Func*** funcCatNameConteinsU = ***;
             //Func*** funcCatIsDomesticAndWhite = ***;
 
```
//write result to variable

Where is your result? Please, add new variable with extension calling.

TheDioBrando replied:
Done

TheDioBrando committed:
Fix code style
Add variables for extension method output
Delete condition for generic type in extension method

TIove commented on src/FuncHW/Program.cs:
```diff
@@ -25,18 +25,25 @@ static void Main()
 
             List<Cat> cats = new() { barsik, pushok, murzik, dymok, murka };
 
-            //Func*** funcPersonIsActive = ***;
```
Rollback this comments, please

You can delete only commented-functions that you implemented

TIove replied:
This is necessary, because, when your merge request will be merged, all other students will not see theirs tasks

TheDioBrando replied:
Sorry, returned comments

TheDioBrando committed:
Returned comments of unrealized functionality

MorozVl reviewed (commented):
LGTM
